selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/requestfactory/TestDefaultUserHttpJsonRequestFactory.java

```diff
@@ -10,22 +10,37 @@
  */
 package org.eclipse.che.selenium.core.requestfactory;
 
+import static java.lang.String.format;
+
 import com.google.inject.Inject;
-import com.google.inject.Provider;
 import com.google.inject.Singleton;
-import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.client.TestAuthServiceClient;
+import org.eclipse.che.selenium.core.user.TestUser;
 
-/** @author Dmytro Nochevnov */
+/**
+ * @author Dmytro Nochevnov
+ * @author Anton Korneta
+ */
 @Singleton
 public class TestDefaultUserHttpJsonRequestFactory extends TestHttpJsonRequestFactory {
-  private Provider<DefaultTestUser> testUserProvider;
+
+  private final TestUser testUser;
+  private final TestAuthServiceClient authServiceClient;
 
   @Inject
-  public TestDefaultUserHttpJsonRequestFactory(Provider<DefaultTestUser> testUserProvider) {
-    this.testUserProvider = testUserProvider;
+  public TestDefaultUserHttpJsonRequestFactory(
+      TestAuthServiceClient authServiceClient, TestUser testUser) {
+    this.testUser = testUser;
+    this.authServiceClient = authServiceClient;
   }
 
+  @Override
   protected String getAuthToken() {
-    return testUserProvider.get().getAuthToken();
+    try {
+      return authServiceClient.login(testUser.getEmail(), testUser.getPassword());
+    } catch (Exception ex) {
+      throw new RuntimeException(
+          format("Failed to access get token for user '%s'", testUser.getName()));
```

Comment:
It's better to log user's email for the full information so as user's name is usual just the part of email before "@"